application.js

```diff
@@ -39,15 +39,30 @@ function playPuppies(puppies) {
     }
 
     function showPuppy(newPup) {
-        var url = makeURL(newPup);
-        $('#puppy > source').attr('src', url);
+        var url;
+
+        if (isMobileOrTablet()) {
+            url = makeURL(newPup, 'gif');
+            $('#puppygif').remove();
+            setTimeout(function() {
+                $('<img id="puppygif" class="centered" src="' + url + '">').insertAfter('#puppy');
+            }, 500);
```

Comment:
* New element forces IOS to reload the next gif immediately
* Timeout is needed for Android and IOS to load the gif once the element has been added.